core/src/main/java/org/elasticsearch/search/aggregations/AbstractAggregationBuilder.java

```diff
@@ -118,7 +118,7 @@ public AB setMetaData(Map<String, Object> metaData) {
 
     @Override
     public Map<String, Object> getMetaData() {
-        return Collections.unmodifiableMap(metaData);
+        return metaData == null ? Collections.emptyMap() : Collections.unmodifiableMap(metaData);
```

Comment:
its fixing a bug as the metadata was always null if not set, we just never called `getMetadata()` from within Elasticsearch with an unset metadata so we didn't see the bug. I agree that ideally the metadata would never be null but I'm not sure on the side effects of that change so I think it would be better to explore that in a different PR. Does that work for you?